botocore/credentials.py

```diff
@@ -1005,19 +1063,24 @@ class SharedCredentialProvider(CredentialProvider):
 
     ACCESS_KEY = 'aws_access_key_id'
     SECRET_KEY = 'aws_secret_access_key'
+    MFA_SERIAL = 'mfa_serial'
     # Same deal as the EnvProvider above.  Botocore originally supported
     # aws_security_token, but the SDKs are standardizing on aws_session_token
     # so we support both.
     TOKENS = ['aws_security_token', 'aws_session_token']
 
-    def __init__(self, creds_filename, profile_name=None, ini_parser=None):
+    def __init__(self, creds_filename, profile_name=None, ini_parser=None,
+                 prompter=getpass.getpass, cache=None, client_creator=None):
         self._creds_filename = creds_filename
         if profile_name is None:
             profile_name = 'default'
         self._profile_name = profile_name
         if ini_parser is None:
             ini_parser = botocore.configloader.raw_config_parse
         self._ini_parser = ini_parser
+        self._prompter = prompter
+        self.cache = cache
```

Comment:
need it public to perform similar to https://github.com/aws/aws-cli/blob/develop/awscli/customizations/assumerole.py#L41